src/main/java/bio/terra/pdao/bigquery/BigQueryPdao.java

```diff
@@ -481,54 +478,42 @@ private void storeRowIdsForRoot(String studyDatasetName,
-        InsertAllRequest.Builder requestBuilder = InsertAllRequest.newBuilder(bqTableId);
+        StringBuilder builder = new StringBuilder();
+        builder.append("INSERT INTO `")
+            .append(projectId).append('.').append(datasetName).append('.').append(PDAO_ROW_ID_TABLE)
+            .append("` (").append(PDAO_TABLE_ID_COLUMN).append(",").append(PDAO_ROW_ID_COLUMN)
+            .append(") SELECT ").append("'").append(tableId).append("' AS ").append(PDAO_TABLE_ID_COLUMN)
+            .append(", T.rowid AS ").append(PDAO_ROW_ID_COLUMN)
+            .append(" FROM (SELECT rowid FROM UNNEST([");
+
+        // Put all of the rowids into an array that is unnested into a table
+        String prefix = "";
         for (String rowId : rowIds) {
-            Map<String, Object> row = new HashMap<>();
-            row.put(PDAO_TABLE_ID_COLUMN, tableId);
-            row.put(PDAO_ROW_ID_COLUMN, rowId);
-            requestBuilder.addRow(row);
-        }
-        InsertAllRequest request = requestBuilder.build();
-
-        InsertAllResponse response = bigQuery.insertAll(request);
-        if (response.hasErrors()) {
-            logger.error("Row Id InsertAll had errors:");
-            for (List<BigQueryError> errorList : response.getInsertErrors().values()) {
-                for (BigQueryError bigQueryError : errorList) {
-                    logger.debug("    {}", bigQueryError);
-                }
-            }
+            builder.append(prefix).append("'").append(rowId).append("'");
+            prefix = ",";
         }
 
-        // DEBUG: dump the row id table
-        debugDumpRowIdTable(datasetName);
-    }
-
-    private void debugDumpRowIdTable(String datasetName) {
-        StringBuilder builder = new StringBuilder();
-        builder.append("SELECT ")
-            .append(PDAO_TABLE_ID_COLUMN).append(",").append(PDAO_ROW_ID_COLUMN)
-            .append(" FROM `")
-            .append(projectId).append('.').append(datasetName).append('.').append(PDAO_ROW_ID_TABLE)
-            .append("`");
+        builder.append("]) AS rowid) AS T");
         String sql = builder.toString();
+
         try {
-            logger.info("Dump root row ids");
             QueryJobConfiguration queryConfig = QueryJobConfiguration.newBuilder(sql).build();
-            TableResult result = bigQuery.query(queryConfig);
-            for (FieldValueList row : result.iterateAll()) {
-                logger.info("tableid=" + row.get(0).getStringValue() + "  rowid=" + row.get(1).getStringValue());
-            }
-        } catch (InterruptedException ie) {
-            throw new PdaoException("Debug dump row id query unexpectedly interrupted", ie);
+            bigQuery.query(queryConfig);
+        } catch (InterruptedException e) {
+            throw new IllegalStateException("Insert root row ids unexpectedly interrupted", e);
+        } catch (BigQueryException ex) {
+            logger.error(ex.toString());
```

Comment:
we need to talk about how we handle logging errors. previously we were not going to log when caught and rethrown allowing the outer code to handle it. but we don't seem to be doing a good job of tracking root cause exceptions

Reply:
I've removed the logging. Had it in for debug and forgot to make it go away.